src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresDirectoryDAO.scala

```diff
@@ -449,18 +449,28 @@ class PostgresDirectoryDAO(protected val writeDbRef: DbReference, protected val
     serializableWriteTransaction("setUserAzureB2CId", samRequestContext) { implicit session =>
       val u = UserTable.column
       val results =
-        samsql"""update ${UserTable.table}
+        if (b2cId.value == "")
+          samsql"""update ${UserTable.table}
+                 set (${u.azureB2cId}, ${u.updatedAt}) =
+                 (null,
+                   ${Instant.now()}
+                 )
+                 where ${u.id} = $userId and ${u.googleSubjectId} is not null"""
+            .update()
+            .apply()
+        else
+          samsql"""update ${UserTable.table}
                  set (${u.azureB2cId}, ${u.updatedAt}) =
                  ($b2cId,
                    ${Instant.now()}
                  )
-                 where ${u.id} = $userId and (${u.azureB2cId} is null or ${u.azureB2cId} = $b2cId)"""
-          .update()
-          .apply()
+                 where ${u.id} = $userId and ${u.googleSubjectId} is not null"""
+            .update()
+            .apply()
```

Comment:
I thought about building these queries dynamically, but I like how clear the if/else logic is this way. Let me know what you think.

Reply:
I think it might actually be better to have an `updateSamUser` method that handles updating both AzureB2CIDs and GoogleSubjectIDs. We can have it dynamically build the query based on the fields present in a `SamUser` input